### DISCUSSION TITLE
Surface Book 2 Features

### DISCUSSION THREAD

**Discussion by ckroether** [linux-surface/linux-surface#1074] [Q&A]
I stumbled on this site by accident while looking for digital sketching devices that will work with Linux.
It looks to me like this site has patches, or install packages, that enable Surface Book specific features that do not work with a standard Linux Ubuntu install.  Is that correct?
I am currently running Ubuntu 22.04.2 LTS on my Surface Book 2 but it just works like a standard laptop.  No touchscreen.  No detachable keyboard.
I am new to Linux (as is my husband although he used to be a mainframe programmer).
The information on this site is confusing me.
Can you, someone, assist me with using this information to make my Surface Book more usable again?

Can other pages on this site help me with my Dell Inspirion which also runs Ubuntu?

Thank You
Sorry I am having trouble understanding all of this.  I am trying to learn to use the Terminal and also learning Python.

**Comment by qzed** [comment]
>  It looks to me like this site has patches, or install packages, that enable Surface Book specific features that do not work with a standard Linux Ubuntu install. Is that correct?

Correct. We provide kernel patches, pre-compiled and packaged kernels with those patches applied, and other related packages that improve support for Surface devices. For the Surface Book 2 specifically:
- As you've noticed most of the stuff works out of the box with any fairly recent kernel.
- For detaching the keyboard, you'll need the [surface-dtx-daemon](github.com/linux-surface/surface-dtx-daemon), which is a small user-space program (actually two) that handles detachment requests. This is available in our repos as `surface-dtx-daemon` package. After installation, make sure that the respective services are running by running `systemctl enable --now surface-dtx-daemon.service` and `systemctl enable --user --now surface-dtx-userd.service`.
- For touch, you'll need a patched kernel and the `iptsd` package from our repos.
- For cameras, you'll also need a patched kernel and `libcamera`. See https://github.com/linux-surface/linux-surface/wiki/Camera-Support#libcamera-support for a bit more details on that. This is still work-in-progress.
- The dGPU should work out of the box, provided you install the `nvidia-dkms` package from the original Ubuntu repos.

You can follow the install guide at https://github.com/linux-surface/linux-surface/wiki/Installation-and-Setup. Essentially, there are two steps: 1) Add the repository so `apt` can access it and 2) install the desired packages.

> Can other pages on this site help me with my Dell Inspirion which also runs Ubuntu?

I somewhat doubt that, but maybe. Most things here are quite specific to Surface devices but some quirks may also apply to other devices. Unfortunately, I do not know anything about the Dell Inspiron, so I can't say definitively.

**Reply by ckroether** [reply]
Thank you for your reply.  I will these patches tomorrow.

**Reply by Isazcode** [reply]
did all the patches work for you?
